src/Fixer/FunctionNotation/UseArrowFunctionsFixer.php

```diff
@@ -0,0 +1,214 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70400 && $tokens->isAllTokenKindsFound([T_FUNCTION, T_RETURN]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $analyzer = new TokensAnalyzer($tokens);
+
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION) || !$analyzer->isLambda($index)) {
+                continue;
+            }
+
+            // Find parameters end
+            // Abort if they are multinlined
+
+            $parametersStart = $tokens->getNextMeaningfulToken($index);
+
+            if ($tokens[$parametersStart]->isGivenKind(CT::T_RETURN_REF)) {
+                $parametersStart = $tokens->getNextMeaningfulToken($parametersStart);
+            }
+
+            $parametersEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $parametersStart);
+
+            if ($this->isMultilined($tokens, $parametersStart, $parametersEnd)) {
+                continue;
+            }
+
+            // Find `use ()` start and end
+            // Abort if it contains reference variables
+
+            $next = $tokens->getNextMeaningfulToken($parametersEnd);
+
+            $useStart = null;
+            $useEnd = null;
+
+            if ($tokens[$next]->isGivenKind(CT::T_USE_LAMBDA)) {
+                $useStart = $next;
+
+                if ($tokens[$useStart - 1]->isGivenKind(T_WHITESPACE)) {
+                    --$useStart;
+                }
+
+                $next = $tokens->getNextMeaningfulToken($next);
+
+                while (!$tokens[$next]->equals(')')) {
+                    if ($tokens[$next]->equals('&')) {
+                        // varibles used by reference are not supported by arrow functions
+                        continue 2;
+                    }
+
+                    $next = $tokens->getNextMeaningfulToken($next);
+                }
+
+                $useEnd = $next;
+                $next = $tokens->getNextMeaningfulToken($next);
+            }
+
+            // Find opening brace and following `return`
+            // Abort if there is more than whitespace between them (like comments)
+
+            $braceOpen = $tokens[$next]->equals('{') ? $next : $tokens->getNextTokenOfKind($next, ['{']);
+            $return = $braceOpen + 1;
+
+            if ($tokens[$return]->isGivenKind(T_WHITESPACE)) {
+                ++$return;
+            }
+
+            if (!$tokens[$return]->isGivenKind(T_RETURN)) {
+                continue;
+            }
+
+            // Find semicolon of `return` statement
+
+            $semicolon = $tokens->getNextTokenOfKind($return, ['{', ';']);
+
+            if (!$tokens[$semicolon]->equals(';')) {
+                continue;
```

Comment:
this is not covered by the unit test, can you take a look?

Reply:
good catch!
fixed: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4778/commits/cfb764f0a217af0f1a84e3b539638d43b8b24419